File: tela.c
#include <stdio.h>
#include <stdlib.h>
#include <conio.h>
#include "tela.h"

void limparTela() {
    system("cls");
}

void esperarEnter() {

    printf("\n              Pressione Enter para continuar...");

    int tecla;

    do {
        tecla = _getch();
    } while (tecla != 13);
}

void mostrarTexto(
    const char *linha1,
    const char *linha2,
    const char *linha3
) {

    limparTela();

    printf("\n");
    printf("====================================================\n");
    printf("|                                                  \n");
    printf("|  %s\n", linha1);
    printf("|  %s\n", linha2);
    printf("|  %s\n", linha3);
    printf("|                                                  \n");
    printf("====================================================\n");

    esperarEnter();
}

File: Tela.java
import java.io.IOException;
import java.util.Scanner;

public class Tela {

    public static final Scanner SC = new Scanner(System.in);

    public static void limparTela() {
        try {
            if (System.getProperty("os.name").toLowerCase().contains("windows")) {

                new ProcessBuilder("cmd", "/c", "cls")
                        .inheritIO()
                        .start()
                        .waitFor();

            } else {

                System.out.print("\033[H\033[2J");
                System.out.flush();
            }

        } catch (IOException | InterruptedException e) {
            System.out.println("\n".repeat(50));
        }
    }

    public static void esperarEnter() {
        System.out.print(
            "\n              Pressione Enter para continuar..."
        );

        SC.nextLine();
    }

    public static int lerInteiro() {

        while (true) {

            String entrada = SC.nextLine();

            try {
                return Integer.parseInt(entrada.trim());

            } catch (NumberFormatException e) {
                System.out.print("Digite um número válido: ");
            }
        }
    }

    public static void mostrarTexto(
            String linha1,
            String linha2,
            String linha3) {

        limparTela();

        System.out.println();
        System.out.println(
            "===================================================="
        );
        System.out.println("|");
        System.out.println("|  " + linha1);
        System.out.println("|  " + linha2);
        System.out.println("|  " + linha3);
        System.out.println("|");
        System.out.println(
            "===================================================="
        );

        esperarEnter();
    }
}